crates/ruff_notebook/src/notebook.rs

```diff
@@ -80,14 +80,127 @@ impl Cell {
         // Ignore cells containing cell magic as they act on the entire cell
         // as compared to line magic which acts on a single line.
         !match source {
-            SourceValue::String(string) => string
-                .lines()
-                .any(|line| line.trim_start().starts_with("%%")),
+            SourceValue::String(string) => string.lines().any(Self::is_magic_cell),
             SourceValue::StringArray(string_array) => string_array
                 .iter()
-                .any(|line| line.trim_start().starts_with("%%")),
+                .map(String::as_str)
+                .any(Self::is_magic_cell),
         }
     }
+
+    /// Returns `true` if a cell should be ignored due to the use of cell magics.
+    fn is_magic_cell(line: &str) -> bool {
+        let line = line.trim_start();
+
+        // Detect cell magics (which operate on multiple lines).
+        if line.starts_with("%%") {
+            return true;
+        }
+
+        // Detect automatic line magics (automagic), which aren't supported by the parser. If a line
+        // magic uses automagic, Jupyter doesn't allow following it with non-magic lines anyway, so
+        // we aren't missing out on any valid Python code.
+        //
+        // For example, this is valid:
+        // ```jupyter
+        // cat /path/to/file
+        // cat /path/to/file
+        // ```
+        //
+        // But this is invalid:
+        // ```jupyter
+        // cat /path/to/file
+        // x = 1
+        // ```
+        //
+        // See: https://ipython.readthedocs.io/en/stable/interactive/magics.html
+        if line.split_whitespace().next().is_some_and(|token| {
+            matches!(
+                token,
+                "alias"
```

Comment:
There are shell commands listed here as well. Can we add a comment on the source of the possible shell commands here?

Separately, does this list include _all_ of the line and cell magics in the linked documentation (https://ipython.readthedocs.io/en/stable/interactive/magics.html)?

Reply:
I believe this only applies to line magics, and not cell magics.

Reply:
Please correct me if I'm wrong though.

Reply:
Yes, you're correct.